app/controllers/planning_applications/neighbour_letter_batches_controller.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class PlanningApplications::NeighbourLetterBatchesController < AuthenticationController
+  def index
+    set_planning_application
+
+    respond_to do |format|
+      format.csv do
+        result = []
+        result << "address,batch,date"
+        @planning_application.consultation.neighbour_letter_batches.each_with_index do |batch, i|
+          batch.neighbour_letters.each do |letter|
+            result << %("#{letter.neighbour.address}",Neighbour letter #{i + 1},#{batch.created_at})
+          end
+        end
+
+        render plain: result.join("\n") + "\n", layout: false
```

Comment:
Why not use the CSV library and a ruby template to do this like we do on Petitions:
https://github.com/alphagov/e-petitions/blob/master/app/views/petitions/index.csv.ruby